src/Sylius/Bundle/CoreBundle/DependencyInjection/Compiler/BackwardsCompatibility/Symfony6PrivateServicesPass.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\DependencyInjection\Compiler\BackwardsCompatibility;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/** @internal */
+final class Symfony6PrivateServicesPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        // Some services are still fetched directly from the container in both Sylius and its extracted bundles (e.g. ResourceBundle)
+        // Before we fix it let's make them public to keep backward compatibility
+
+        $this->makeServicePublic('form.factory', $container);
+        $this->makeServicePublic('security.authorization_checker', $container);
+        $this->makeServicePublic('security.csrf.token_manager', $container);
+        $this->makeServicePublic('security.token_storage', $container);
+        $this->makeServicePublic('validator', $container);
+
+        if (str_starts_with((string) $container->getParameter('kernel.environment'), 'test')) {
+            $this->makeServicePublic('filesystem', $container);
+            $this->makeServicePublic('session.factory', $container);
+            $this->makeServicePublic('sylius.command_bus', $container);
+        }
```

Comment:
A slightly different topic - these services are used in our PHPUnit tests and, as far as I know, they _have to_ be taken directly from the container. Therefore I make them public in the test (and test_cached) environments 🚀

Reply:
Are you sure, on PHPUnit a test can be retrieved from the container even if it's private, it's not the same container on PHPUnit tests

Reply:
On KernelTestCase:

```
try {
            return self::$kernelContainer->get('test.service_container');
        } catch (ServiceNotFoundException $e) {
            throw new \LogicException('Could not find service "test.service_container". Try updating the "framework.test" config to "true".', 0, $e);
        }
```

Reply:
> The only remaining drawback of "private services by default" is that testing was harder than before. Some developers even defined some config in the test environment to make all services public in tests. **In Symfony 4.1, we did the same and now tests allow fetching private services by default.**

Afaik there's no need to mark services as public for testing purposes 🤔.

ref: https://symfony.com/blog/new-in-symfony-4-1-simpler-service-testing

Reply:
I was 100% sure it was failing and I was also surprised 😱 https://github.com/Sylius/Sylius/pull/14407 fix proposed